docs/network-policy.md

```diff
@@ -0,0 +1,361 @@
+# NetworkPolicy
+
+Kubernetes NetworkPolicy documentation: https://kubernetes.io/docs/concepts/services-networking/network-policies
+
+Kubernetes NetworkPolicy API reference: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#networkpolicy-v1-networking-k8s-io
+
+By default the network traffic from and to K8s pods is not restricted in any way. Using Network Policy is a way to enforce network isolation of selected pods. When a pod is selected by a NetworkPolicy allowed traffic is specified by the `ingress` and `egress` sections.  
+
+Each NetworkPolicy object consists of four sections:
+
+1. `podSelector`: a label selector that determines which pods the NetworkPolicy applies to
+2. `policyTypes`: determines which policy tpes are included, if none are selected then `Ingress` will always be set and `egress` will be set if any `egress` rules are applied 
+3. `ingress rules`: determines the sources that pods selected by the ingress rule can receive traffic from 
+4. `egress rules`: determines the sinks that pods selected by the egress rule can send trafic to 
+
+# NetworkPolicy features 
+
+These are descibed in order and are additive 
+
+## **Unicast default-deny**
+
+When a pod is selected by one or more NetworkPolicies, the `policyTypes` is set to both `ingress` and `egress`, and if no rules are specified it becomes isolated and all unicast ingress and egress traffic is blocked. 
+
+```
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: default-deny-all
+spec:
+  podSelector: {}
+  policyTypes:
+  - Ingress
+  - Egress
+```
+
+If only ingress traffic to all pods in a namespace needs to be blocked the following can be used 
```

Comment:
it might be worth mentioning that these apply to all pods in the same namespace as the networkPolicy as opposed to some other namespace